sklearn/metrics/_scorer.py

```diff
@@ -453,31 +340,11 @@ def _score(self, method_caller, clf, X, y, **kwargs):
             kwargs=kwargs,
         )
 
-        y_type = type_of_target(y)
-        if y_type not in ("binary", "multilabel-indicator"):
-            raise ValueError("{0} format is not supported".format(y_type))
-
-        if is_regressor(clf):
-            y_pred = method_caller(clf, "predict", X)
-        else:
-            pos_label = self._get_pos_label()
-            try:
-                y_pred = method_caller(clf, "decision_function", X, pos_label=pos_label)
-
-                if isinstance(y_pred, list):
-                    # For multi-output multi-class estimator
-                    y_pred = np.vstack([p for p in y_pred]).T
-
-            except (NotImplementedError, AttributeError):
-                y_pred = method_caller(clf, "predict_proba", X, pos_label=pos_label)
-                if isinstance(y_pred, list):
-                    y_pred = np.vstack([p[:, -1] for p in y_pred]).T
+        pos_label = None if is_regressor(estimator) else self._get_pos_label()
+        y_pred = method_caller(estimator, self._response_method, X, pos_label=pos_label)
```

Comment:
The main change the `_Scorer`s is here. We need a single class and leverage the `_get_response_values` (hidden within the `method_caller`).